llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -8313,19 +8265,109 @@ const Vkgc::PipelineOptions *SPIRVToLLVM::getPipelineOptions() const {
   return static_cast<Llpc::Context *>(m_context)->getPipelineContext()->getPipelineOptions();
 }
 
-// This method does preparation work for adding the scratch access bounds checks for memory operations
-// like OpLoads and OpStores. We are handling private and function memory classes here.
-// The idea behind this method is that while translating a OpLoad or OpStore which loads from local scratch memory using
-// a unknown (run-time) index into an array, we parse the operands of the corresponding access chain, add a chain of
-// ANDed compares to check if the bounds requirements are met and then do a conditional jump to either the intermediate
-// block which handles the memory operation or the final block (which will skip the memory op, if bounds checks failed).
-// As mentioned, this does preparation work like inserting the bounds checks, its up to the caller caring about
-// inserting the OpLoad or OpStore. This also sets a flag that signals the main translation loop that we want to
-// continue inserting subsequent instructions at the final block instead of just going back to the parent block.
-// Currently, OpLoads will yield zero and OpStores will just omit the store, if bounds checks failed. This is up to the
-// caller. This implementation handles nested structs using arrays and 1...n-dimensional arrays. This returns if we need
-// to insert a bounds checks and pointers to the newly created basic blocks (which can be optimized away by subsequent
-// passes, if not needed).
+bool SPIRVToLLVM::scratchBoundsChecksEnabled() const {
```

Comment:
For the new functions added, I think we can follow our coding style giving them separate lines,, header comments, and parameter comments. Indeed, there are mixed styles in SPIR-V translator and we need time to correct all of them.